packages/react/types/index.d.ts

```diff
@@ -116,9 +116,8 @@ export namespace jsx {
     interface ElementChildrenAttribute
       extends ReactJSXElementChildrenAttribute {}
 
-    type LibraryManagedAttributes<C, P> = C extends React.ComponentType<infer T>
-      ? WithConditionalCSSProp<T>
-      : WithConditionalCSSProp<ReactJSXLibraryManagedAttributes<C, P>>
+    type LibraryManagedAttributes<C, P> = WithConditionalCSSProp<P> &
+      ReactJSXLibraryManagedAttributes<C, P>
```

Comment:
Huh, I was under impression that `P` stands for props passed to the JSX element, and not for all props of `C`. It seems that the compiler resolves those on its own so my `React.ComponentType<infer T>` was completely redundant - and it even has introduced this problem with `children`. 

I was curious to see how this `LibraryManagedAttributes` works and I've prepared a TS playground to test various scenarios, might be interesting for somebody: [TS playground](https://www.typescriptlang.org/play/index.html?jsx=2&ssl=32&ssc=30&pln=32&pc=47#code/PQKhAIAECsGcA9x0SYAoAlgWwA4HsAnAF3AgENZwAlAUzIGMSAzAvLcAcgLsY7TSYBXAHaMMeYUgQAKAJTgA3uAC+aYWSw1YOBjSmIFacOBrx8xcOs3bd4AFIBlABqKjxk2cIkiATxx6AGQwAIwIyAh8AWTJ1AHMaABMAQSIiAhDBIi0AHgBhABpwAAUAPnAAXmLwADJFcHoAfgAucALwHGaq1WNVVQERMQlWthxpHFYcWBalJjw8FuFBLGCaAkLg8M7YNIxhWJV5Q2NuIkECSUWAG0u0PoBucDyR8Fm8coUAFgAmZXr3q8uvxw-0E11+wBK-HoEm24EiNCweESuWelVoDCIADpNIjpCjcLJ+A9svDEcjnq93t9fvQQWD2nTAeAIVCYSR8TgAOoYIgACwAkrhLhh6DzcryMJcEtxhC10YxMQAxXLZGZzBZLFZrcAbAhbHZ7FRlSpCUREcSSDljCawQ5uE5nC6gm59YzEjncvmCnDC0VEcWS6U0SSUz4-P4KAFAxngyFoaHCWEenm8gAiNCYZFBREolWteEm0xe6ssmtW63CLW26UNynk5TKR3ADvOpeutzQyb56cz2dgmISGazlyIRRtFTqupaHHUs7IHBURMeXbTQ77xbeYZpMYZkedsaAA)

I was also curious why there is a difference between both of those and I think that is because the TS compiler gets `P` from a callable signature, which is defined here:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0f095853ae6233ad5482a7204dab9326b128ad5c/types/react/index.d.ts#L547
but `React.ComponentType` infers a generic given to `FunctionComponent` which is defined here:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0f095853ae6233ad5482a7204dab9326b128ad5c/types/react/index.d.ts#L546
and thus it doesn't have those implicit `children`.

As to the proposed change - there is probably no real difference between what u got and this, but maybe this would look slightly more naturally?
```suggestion
    type LibraryManagedAttributes<C, P> = WithConditionalCSSProp<ReactJSXLibraryManagedAttributes<C, P>>
```

Reply:
I originally did that, though that doesn't seem to add the css prop for components: https://www.typescriptlang.org/play/index.html?jsx=1&ssl=1&ssc=1&pln=50&pc=25#code/PQKhAIAECsGcA9x0SYAoAlgWwA4HsAnAF3AG9wBjAgUwEMjqBRAG2q2oDsSBfcAMwJ4s4AOQ1aFIiLRoiATxzVwAdQxEAFgGE8HACZqMO2s00BlUwAVBOADwWAfOAC8ois1qxYAOVrsR4angGPVhwAGtqOTw+cAs0cHAAfnAACgsAoM5dUPI3D29fakSALnBYIgIMDgBzcF5k9IAyMkpPEvA8ACNoakk68FKLAEp4gdiZYGBwAFcOHAkw8GNmcHVqGnAiPCWANzwMXXAqviq1JVhqZj4AWho+dc4KKtqAdzWOcF1qE45njumCB0Xh8AFKmAAa4A4hVg8wo1FkCiUACU6JIweCWGxOCQXBiAHRY9hcRGKcCoiREDFEnGadyeZzgAk0rh0-KklFoqkQllEACCRAqGE60wYsCseEUxDkjOZrGJ-MFlRFYolUvkHPJXOp8tp6gwzF0NA4AqFKqUeIhhN1rP1huNpuVooR8jJFPREIAMsKCLQCHIALK0aHVai6R3C52wGyaAA0sUclvB+O9nV9-qDIbDEfN0bjCc17u54IAklxKhxYBgKDmo7KrWWhZXq7WxYXtRDGxWqxQ2Z5W9RowAVRNMhvlqo9vuwAfD+ztykYruT6u80JJ-HL5sUNcyL55DYUHTlJAIUqu6jRSjiBi8tD79wbaHsWESJTIMijKEwuFKDGfhIEiqBgCD4N9wF5DJgmyLVFx5G0SFIbgvyOLh1jA+EIIQ6coKyUIix1bFWXpHJkMA1CQIwpReVnNV1g1ciEkCaD8I7TEENo6x6JlJCUOA9DwN5TQ7SNTgBxQpjMhCWCPXYoiiGEg1RJNJVIwYMhkJQi9wFTdNA2DWhQ3DVTcxjeMHEZVQNG0PQDCMExzDVGwCK9H0-X0rNjLNKMzITedGPAGRyP40DwK3HtZ1w6SXNLCdt0i3jgrQ0LMPC6tp1nGwRwkqKYJitLexIzKRw0oLAO0gq10ZUgcoAbQAaVQ8JIivfK4p7NcAF1SjapsOoQ2AGs68BmhqgKEgoNpSi6HpJBysjAIW5DNIAbhsfQdlaWAnFIJDeGAewVpkNA+FmSRDA+AM5G0XAdBxFIcGsWBilIPJPB8dgSnKCtqm4IYxvAGgiABD4OGmZhmDQbgjrQGwrpu-AOBxLadr28ADqAA

Reply:
Oh wait - If you distribute over props, it works([playground](https://www.typescriptlang.org/play/index.html?jsx=1&ssl=50&ssc=25&pln=49&pc=1#code/PQKhAIAECsGcA9x0SYAoAlgWwA4HsAnAF3AG9wBjAgUwEMjqBRAG2q2oDsSBfcAMwJ4s4AOQ1aFIiLRoiATxzVwAdQxEAFgGE8HACZqMO2s00BlUwAVBOADwWAfOAC8ois1qxYAOVrsR4angGPVhwAGtqOTw+cAs0cHAAfnAACgsAoM5dUPI3D29fakSALnBYIgIMDgBzcF5k9IAyMkpPEvA8ACNoakk68FKLAEp4gdiZYGBwAFcOHAkw8GNmcHVqGnAiPCWANzwMXXAqviq1JVhqZj4AWho+dc4KKtqAdzWOcF1qE45njumCB0Xh8AFKmAAa4A4hVg8wo1FkCiUACU6JIweCWGxOCQXBiAHRY9hcRGKcCoiREDFEnGadyeZzgAk0rh0-KklFoqkQllEACCRAqGE60wYsCseEUxDkjOZrGJ-MFlRFYolUvkHPJXOp8tp6gwzF0NA4AqFKqUeIhhN1rP1huNpuVooR8jJFPREIAMsKCLQCHIALK0aHVai6R3C52wGyaAA0sUclvB+O9nV9-qDIbDEfN0bjCc17u54IAklxKhxYBgKDmo7KrWWhZXq7WxYXtRDGxWqxQ2Z5W9RowAVRNMhvlqo9vuwAfD+ztykYruT6u80JJ-HL5sUNcyL55DYUHTlJAIUqu6jRSjiBi8tD79wbaHsWESJTIMijKEwuFKDGfhIEiqBgCD4N9wF5DJgmyLVFx5G0SFIbgvyOLh1jA+EIIQ6coKyUIix1bFWXpHJkMA1CQIwpReVnNV1g1ciEkCaD8I7TEENo6x6JlJCUOA9DwN5TQ7SNTgBxQpjMhCWCPXYoiiGEg1RJNJVIwYMhkJQi9wFTdNA2DWhQ3DVTcxjeMHEZdJmLwpYOBlZICK9H0-X0rNjLNKMzJUNQtB0fQiEMaETHMNU7Hsexig4agdnWCTwBkcj+NA8Ctx7WdcOkxzSwnbd0t4xK0OSzDUuradZxsEc4uszK2JK3sSPKkcNISwDtLqtdGVIOKAG0AGlUPCSIryy9qENgABdUoRpyns1z68bwGaLrGMAig2lKLoekkOKyMA3bkM0gBuGx9B2VpYCcUgkN4YB7EOmQ0D4WZJEC8AAzkbRcB0HEUhwaxYGKUg8k8Hx2BKcoK2qbghmW8AaCIAEPg4aZmGYNBuHutAbHez78Cirhzsu67wFuoA)) - I feel like the intersection is kinda better, I would assume an intersection is cheaper than a conditional type?

Reply:
It also works with a simpler:
```ts
```

Reply:
I don't understand why the composition of the types doesn't work here, but if it doesn't then the intersection is fine. 👍